### PR TITLE
fix: Rollback multi-source apps; 2nd follow-up to PR 14124

### PR DESCRIPTION
**Pull request by keithchong**
A bit of history:
Second follow-up change for: https://github.com/argoproj/argo-cd/pull/18615
Original change: https://github.com/argoproj/argo-cd/pull/14124 (See my PR review comments, which led to PR 18615)

The rollback and history page is not user-friendly for multiple sources (at least at the time of the original implementation earlier this year).  A lot of scrolling is needed.  In fact, a similar usability [issue](https://github.com/argoproj/argo-cd/issues/12183) was brought up even for single-source apps, which was addressed by https://github.com/argoproj/argo-cd/pull/16871, by adding a Show/Hide button.  It however has an issue where for multi-source apps, the single button would show or hide _all_ the source parameter sections, and it should actually show/hide each section individually. This PR fixes that problem and also makes it look a bit more consistent with the Sources tab for multi-source apps, by using collapsible sections.

There is no Issue opened for this.

See screenshots and animated gif below.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

Updated to Rollback and History panel, showing multi-source app history:

CollapsibleSections

The current Sources Tab for a multi-source app which has collapsible sections, for comparison:

SourceTabMultisourceApp

Single Source collapsed (no Show/Hide button), to be consistent with multi-source case:

SingleSourceCollapsed

Single Source expanded:

SingleSourceExpanded

Error loading, when expanding section:

20241026-ExpandedError

Demo animated gif showing the changes for a multi-source app:

DemoRollbackHistoryMultiSourceApp

### CODE REVIEW

**Comment by keithchong** [comment]
Hi @pasha-codefresh , @reggie-k , because you reviewed https://github.com/argoproj/argo-cd/pull/16871 , could you please review this PR change? This PR will remove the Show/Hide button.

**Comment by pasha-codefresh** [comment]
Sure @keithchong , we will allocate time and will review it

**Review by ishitasequeira — APPROVED** [review]
Overall change looks good to me!!
Thanks @keithchong for addressing this.

**Review by ishitasequeira — COMMENTED** [review]
@keithchong looks like UI lint is still failing. Can you take a look at that?

File: ui/src/app/applications/components/application-deployment-history/application-deployment-history-details.tsx [hunk 1]
```diff
@@ -87,35 +166,60 @@ export const ApplicationDeploymentHistoryDetails = ({app, info, index}: props) =
...
+                                            versionId: recentDeployments[index].id,
+                                            appName: app.metadata.name
+                                        }}
+                                        load={src => services.repos.appDetails(src, src.appName, app.spec.project, i, recentDeployments[index].id)}>
+                                        {(details: models.RepoAppDetails) => (
+                                            <React.Fragment>
+                                                <div
+                                                    id={'floating_title_' + i}
+                                                    className='editable-panel__sticky-title'>
+                                                    <div style={{marginTop: '20px'}}>
```
Comment (todaywasawesome) [inline ui/src/app/applications/components/application-deployment-history/application-deployment-history-details.tsx:200]:
```suggestion
                                        {(details: models.RepoAppDetails) => (
                                            <React.Fragment>
                                                <div id={'floating_title_' + i} className='editable-panel__sticky-title'>
                                                    <div style={{marginTop: '20px'}}>
```

Linter doesn't like multi-line divs

**Review by todaywasawesome — DISMISSED** [review]
Just the lint fix

**Comment by pasha-codefresh** [comment]
@CodiumAI-Agent /review

File: ui/src/app/applications/components/application-deployment-history/application-deployment-history-details.tsx [hunk 2]
```diff
@@ -43,36 +104,54 @@ export const ApplicationDeploymentHistoryDetails = ({app, info, index}: props) =
...
-                                </div>
-                            )}
-                        </DataLoader>
+                    {showParameterDetails ? (
+                        <div
+                            id={`'history-expanded'`}
+                            key={`'history-expanded'`}
+                            className={classNames('white-box')}
+                            style={{
+                                marginTop: singleSourceTopMargin,
```
Comment (pasha-codefresh) [inline ui/src/app/applications/components/application-deployment-history/application-deployment-history-details.tsx:114]: do you think we can move it to styles ? or it is generic practice in ArgoCD ui?

**Review by reggie-k — COMMENTED** [review]
Thanks a lot for this fix, @keithchong!

I have tested manually and this panel looks much better than the button and better integrates with the ArgoCD UI overall.

One thing I liked about the button was that I could press on exactly the same location on the screen to both show and hide, while with the new panel I could click anywhere on the panel it to expand it, but can only click on the arrow to collapse, and I also think that the arrow needs to be a bit more visible in dark mode.

WDYT?

File: ui/src/app/applications/components/application-deployment-history/application-deployment-history-details.tsx (hunk 2 above)
Comment (keithchong) [reply ui/src/app/applications/components/application-deployment-history/application-deployment-history-details.tsx:114]: It should be in the css file.  I had a variable there `singleSourceTopMargin`. I going to remove it and move these styles out.

**Comment by keithchong** [comment]
> Thanks a lot for this fix, @keithchong!
> 
> I have tested manually and this panel looks much better than the button and better integrates with the ArgoCD UI overall.
> 
> One thing I liked about the button was that I could press on exactly the same location on the screen to both show and hide, while with the new panel I could click anywhere on the panel it to expand it, but can only click on the arrow to collapse, and I also think that the arrow needs to be a bit more visible in dark mode.
> 
> WDYT?

Here is the reason why I have the expand/collapse behaviour this way.

1. The collapsed version of these sections look pretty much the same as those cards shown in the Settings page. I wanted to have the same look and feel. However, instead of the arrow pointing to the right as in the Settings page, these arrows will be pointing down, to indicate the collapsed state.  In the Settings page, you can also click any where on the card to bring you to the next page.  Here, for the collapsed section, I wanted to maintain that behaviour.  However, we could just as easily force the user to click on the drop-down arrow too, but the current way makes it easier. (Note that if we were using a standard framework, like Patternfly or Carbon, we would simply just use whatever widget they provide.)
2. After the section is expanded, you must click on the up-arrow to collapse it, as you've discovered. This must be the case because users may want to copy the field values, so they must be able to click to select the text (instead of collapsing the section).

The arrows are currently the same colour in dark mode as those arrows in the Settings page. i agree that it can be brighter:

Collapsed:
20241026-DarkModeCollapsed

Expanded:
20241026-DarkModeExpanded

File: ui/src/app/applications/components/application-deployment-history/application-deployment-history-details.tsx (hunk 1 above)
Comment (keithchong) [reply ui/src/app/applications/components/application-deployment-history/application-deployment-history-details.tsx:200]: Thanks for making these changes. To make things ever-so-slightly-simpler for me, so I don't have to pull down your commit and merge it with my new changes, I just made the change in my workspace.

**Comment by keithchong** [comment]
@pasha-codefresh , @reggie-k - updated and rebased. Thanks for your time again.

**Comment by reggie-k** [comment]
Thanks for the making the arrow look better in dark mode, Keith.
So I understand we cannot make it both collapsible and text-selectable?
Actually, expanding/collapsing the "SYNC STATUS" on the left filters sidebar is available by clicking on the arrow only and not on the entire row.
I think having the panel collapsible and text searchable would be _nice_, but if it is complicated to achieve, maybe we can enforce collapsing/expanding by clicking the arrow only, like in left filters sidebar, or just leave it as is, up to you.
I don't have a really strong opinion from a UX point of view and this LGTM also as it is right now.

**Review by reggie-k — APPROVED** [review]

**Review by ishitasequeira — APPROVED** [review]
@todaywasawesome any concerns with merging the PR?

**Review by pasha-codefresh — APPROVED** [review]

**Pull request merged**